src/reachy2_sdk/parts/arm.py

```diff
@@ -423,6 +435,149 @@ def send_cartesian_interpolation(
             current_precision_distance_xyz = np.linalg.norm(current_pose[:3, 3] - target[:3, 3])
         self._logger.info(f"l2 xyz distance to goal: {current_precision_distance_xyz}")
 
+    def _send_linear_interpolation(
+        self,
+        origin_trans: npt.NDArray[np.float64],
+        target_trans: npt.NDArray[np.float64],
+        origin_rot: Quaternion,
+        target_rot: Quaternion,
+        nb_steps: int,
+        time_step: float,
+    ) -> None:
+        """Generate linear interpolation."""
+        for t in np.linspace(0, 1, nb_steps):
+            # Linear interpolation for translation
+            trans_interpolated = (1 - t) * origin_trans + t * target_trans
+
+            # SLERP for rotation interpolation
+            q_interpolated = Quaternion.slerp(origin_rot, target_rot, t)
+            rot_interpolated = q_interpolated.rotation_matrix
+
+            # Recompose the interpolated matrix
+            interpolated_matrix = recompose_matrix(rot_interpolated, trans_interpolated)
+
+            request = ArmCartesianGoal(
+                id=self._part_id,
+                goal_pose=Matrix4x4(data=interpolated_matrix.flatten().tolist()),
+            )
+            self._stub.SendArmCartesianGoal(request)
+            time.sleep(time_step)
+
+    def _send_elliptical_interpolation(
+        self,
+        origin_trans: npt.NDArray[np.float64],
+        target_trans: npt.NDArray[np.float64],
+        origin_rot: Quaternion,
+        target_rot: Quaternion,
+        arc_direction: str,
+        secondary_radius: Optional[float],
+        nb_steps: int,
+        time_step: float,
+    ) -> None:
+        """Generate elliptical interpolation."""
+        vector_target_origin = target_trans - origin_trans
+
+        center = (origin_trans + target_trans) / 2
+        radius = float(np.linalg.norm(vector_target_origin) / 2)
+
+        vector_origin_center = origin_trans - center
+        vector_target_center = target_trans - center
+
+        if radius == 0:
```

Comment:
use np.close here, radius won't likely be exactly 0.
If observed this by asking for send_cartesian_interpolation twice by mistake. I've got the warning "arc direction has no solution"

Reply:
Done